compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2362,7 +2362,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         match path.res {
             Res::Def(DefKind::OpaqueTy, did) => {
                 // Check for desugared `impl Trait`.
-                assert!(ty::is_impl_trait_defn(tcx, did).is_none());
+                assert!(matches!(
+                    ty::impl_trait_origin(tcx, did),
+                    None | Some(hir::OpaqueTyOrigin::FnReturn(_))
+                        | Some(hir::OpaqueTyOrigin::TyAlias)
```

Comment:
why did the TyAlias case change?

Is this reachable for opaque types from other crates or can we avoid checking the `None` case now?